### PR TITLE
Fix B0 for nRF51

### PR DESCRIPTION
**Pull request by oyvindronningstad**
~Blocked by https://github.com/nrfconnect/sdk-zephyr/pull/327~

### CODE REVIEW

**Comment by nvlsianpu** [comment]
please fix the conflict.

File: west.yml [hunk 1]
```diff
@@ -92,7 +92,7 @@ manifest:
     # changes.
     - name: mcuboot
       repo-path: sdk-mcuboot
-      revision: a401d3a056b6ef41a9ead1a7ad5263e028f52549
+      revision: pull/124/head
```
Comment (nvlsianpu) [inline west.yml:95]: update SHA

Comment (oyvindronningstad) [reply west.yml]: Done

**Review by sigvartmh — APPROVED** [review]
Changes look fine to me.

**Comment by nvlsianpu** [comment]
@oyvindronningstad rebase needed.

File: tests/subsys/bootloader/bl_crypto/src/main.c [hunk 2]
```diff
@@ -159,6 +158,8 @@ void test_sha256(void)
 
 	/* Size restrictions. */
 #if CONFIG_FLASH_SIZE > 300
+	test_sha256_string(const_fw_data, ARRAY_SIZE(const_fw_data),
+				image_fw_hash, true);
```
Comment (tejlmand) [inline tests/subsys/bootloader/bl_crypto/src/main.c:162]: I believe that `TAB`s are used for first indentation, and then `spaces` for final alignment of `image_fw_hash` to align with `const_fw_data`.

File: tests/drivers/fprotect/positive/src/main.c [hunk 3]
```diff
@@ -15,7 +15,7 @@
 #include <device.h>
 #include <drivers/flash.h>
 
-static uint8_t write_data[] = "Hello world";
+static uint8_t write_data[] = "HelloWorld";
```
Comment (tejlmand) [inline tests/drivers/fprotect/positive/src/main.c:18]: You added quotes when printing characters, especially to ensure correct handling with spacing.
So wouldn't it be nice to keep the space in `Hello World` so it's ensured it works as expected ?

**Review by hakonfam — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by oyvindronningstad** [comment]
@tejlmand merge? :)

**Pull request merged**